js/output/shared/ast-walker.js

```diff
@@ -23,7 +23,10 @@ window.walkAST = function(node, path, visitors) {
             if (Array.isArray(node[prop])) {
                 let i = 0;
                 while (i < node[prop].length) {
-                    let child = node[prop][i];
+                    // Esprima represents empty slots as null, so set child to
+                    // a dummy node if the initial node is falsy in order to prevent
+                    // a silent crash
```

Comment:
Are there other situations where the node would be false-y besides the empty slot? I'm wondering if it's helpful to have a test case for undefined/null/0.

Reply:
If [modern esprima's type definitions](https://github.com/jquery/esprima/blob/master/src/nodes.ts) are anything to go off of, `null` should be the only falsey value that shows up in node array-properties.  I don't think they'd change their AST structure significantly throughout versions.

Reply:
Okay, thanks, that link clarifies.